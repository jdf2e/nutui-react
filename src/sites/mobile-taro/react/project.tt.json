{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
