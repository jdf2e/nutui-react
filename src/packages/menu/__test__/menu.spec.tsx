import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Success } from '@nutui/icons-react'
import Menu from '@/packages/menu'
import MenuItem from '@/packages/menuitem'

test('should match snapshot', () => {
  const options = [
    { text: '全部商品', value: 0 },
    { text: '新款商品', value: 1 },
    { text: '活动商品', value: 2 },
  ]
  const { asFragment } = render(
    <Menu>
      <MenuItem options={options} value={0} />
    </Menu>
  )
  expect(asFragment()).toMatchSnapshot()
})

test('test props', () => {
  const options = [
    { text: '全部商品', value: 0 },
    { text: '新款商品', value: 1 },
    { text: '活动商品', value: 2 },
  ]
  const testClick = jest.fn((val) => undefined)

  const { container, getByText } = render(
    <Menu lockScroll={false}>
      <MenuItem
        className="custom-className"
        title="custom title"
        options={options}
        value={0}
        disabled
        columns={2}
        icon={<Success />}
        activeTitleClass="activeTitleClass"
        inactiveTitleClass="inactiveTitleClass"
        onChange={testClick}
      />
    </Menu>
  )
  fireEvent.click(getByText('custom title'))

  fireEvent.click(container.querySelectorAll('.nut-menu-container-item ')[1])
  expect(testClick).toBeCalledWith({ text: '新款商品', value: 1 })

  expect(container.querySelector('.custom-className')).toHaveClass(
    'custom-className'
  )
  expect(getByText('custom title')).toHaveTextContent('custom title')
  expect(container.querySelector('.activeTitleClass')).toHaveClass(
    'activeTitleClass'
  )
  expect(container.querySelectorAll('.inactiveTitleClass')[0]).toHaveClass(
    'inactiveTitleClass'
  )
})
