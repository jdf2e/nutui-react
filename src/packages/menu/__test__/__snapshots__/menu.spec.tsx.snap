// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-menu"
  >
    <div
      class="nut-menu-relative"
    >
      <div
        class="nut-menu__bar"
      >
        <div
          class="nut-menu__item "
        >
          <div
            class="nut-menu__title "
          >
            <div
              class="nut-menu__title-text"
            >
              全部商品
            </div>
            <svg
              aria-labelledby="ArrowDown2"
              class="nut-icon nut-icon-ArrowDown2 nut-menu__title-icon"
              role="presentation"
              viewBox="0 0 2472 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1741.19 227.497a121.654 121.654 0 1 1 127.858 206.812L1292.65 790.27a182.482 182.482 0 0 1-191.727 0L524.646 434.31a121.654 121.654 0 1 1 127.859-206.812l544.403 335.887c181.387-111.8 362.773-223.965 544.282-335.887"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="nut-menu-item-container"
      >
        <div
          class="nut-menu-item__wrap"
          style="display: none;"
        >
          <div
            class="nut-menu-item__content"
          >
            <div
              class="nut-menu-item__option active"
              style="flex-basis: 100%;"
            >
              <i>
                <svg
                  aria-labelledby="Check"
                  class="nut-icon nut-icon-Check "
                  color=""
                  role="presentation"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M998.4 245.029c-219.429 153.6-398.629 332.8-552.229 552.228-40.228 58.514-128 54.857-164.571-3.657-69.486-106.057-149.943-186.514-256-256-51.2-32.914-18.286-113.371 40.229-98.743C182.857 460.8 274.286 508.343 358.4 585.143c157.257-190.172 358.4-340.114 588.8-435.2 62.171-25.6 106.057 58.514 51.2 95.086"
                    fill="currentColor"
                    fill-opacity="0.9"
                  />
                </svg>
              </i>
              <div
                class=""
              >
                全部商品
              </div>
            </div>
            <div
              class="nut-menu-item__option "
              style="flex-basis: 100%;"
            >
              <div
                class=""
              >
                新款商品
              </div>
            </div>
            <div
              class="nut-menu-item__option "
              style="flex-basis: 100%;"
            >
              <div
                class=""
              >
                活动商品
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
